core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -93,6 +93,8 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
 
   val retainedStages = conf.getInt("spark.ui.retainedStages", SparkUI.DEFAULT_RETAINED_STAGES)
   val retainedJobs = conf.getInt("spark.ui.retainedJobs", SparkUI.DEFAULT_RETAINED_JOBS)
+  val retainedTasks = conf.getInt("spark.ui.retainedTasks", SparkUI.DEFAULT_RETAINED_TASKS)
```

Comment:
ah good point that hashMap can only hold int number of entries.  

My point was that with taskId being a long and assuming taskId = 0 through X, X here could be a long (2^64) and thus you could have 2^64 number of tasks.  If you can have 2^64 number of tasks then this config should allow you to configure it to retain all of those tasks.

I'm fine with leaving an int.